vaadin-map-flow-parent/vaadin-map-flow/src/main/java/com/vaadin/flow/component/map/configuration/source/VectorSource.java

```diff
@@ -63,10 +63,6 @@ public void removeFeature(Feature feature) {
         removeChild(feature);
     }
 
-    protected static class BaseOptions<T extends BaseOptions<T>>
-            extends Source.BaseOptions<T> {
-    }
-
-    public static class Options extends BaseOptions<Options> {
+    protected static class Options extends Source.Options {
```

Comment:
Looks like a typo was made here:

```suggestion
    public static class Options extends Source.Options {
```

Reply:
I would also suggest adding a couple of tests for `VerctorSource`. It seems there are no tests for it at all.

Reply:
I've fixed the visibility of `VectorSource.Options`, and also updated the visibility of options for abstract classes to protected, as discussed.

Testing this is not as straightforward. It's common practice to put unit tests into the same package as the source file, which also means that the test has access to protected internals of the tested class. I think it would be a bit overblown to add a separate unit test for ensuring the visibility of the options classes, and discovering errors like these is why we have code reviews. However I added missing tests for ensuring that the individual source classes correctly adopt their properties from the options.

Reply:
I meant testing only public classes such as `VectorSource.Options` (if it should be a public class after all).